src/index.ts

```diff
@@ -874,9 +880,18 @@ class ThreadPool {
   }
 
   _maybeDrain () {
-    if (this.taskQueue.size === 0 && this.skipQueue.length === 0) {
+    const totalCapacity = this.options.maxQueue + this.pendingCapacity();
+    const totalQueueSize = this.taskQueue.size + this.skipQueue.length;
+
+    if (totalQueueSize === 0) {
+      this.needsDrain = false;
       this.publicInterface.emit('drain');
     }
+
+    if (totalQueueSize >= totalCapacity) {
+      this.publicInterface.emit('needsDrain');
+      this.needsDrain = true;
```

Comment:
```suggestion
      this.needsDrain = true;
      this.publicInterface.emit('needsDrain');
```

Reply:
Not sure what this event is for?

Reply:
Meant to extend the API only, so you can either listen or check the variable